src/fsharp/TastOps.fs

```diff
@@ -3027,7 +3027,47 @@ let isByrefLikeTy g m ty =
 
 let isSpanLikeTy g m ty =
     isByrefLikeTy g m ty && 
-    not (isByrefTy g ty) 
+    not (isByrefTy g ty)
+
+let isSpanTyconRef (g: TcGlobals) m (tcref: TyconRef) =
+    if isByrefLikeTyconRef g m tcref then
+        match tcref.ILTyconInfo with
+        | TILObjectReprData(_, _, ilTypeDef) -> ilTypeDef.Name = tname_Span
+    else
+        false
+    
+let isSpanTy g m ty =
+    ty |> stripTyEqns g |> (function TType_app(tcref, _) -> isSpanTyconRef g m tcref | _ -> false)
+
+let tryDestSpanTy (g:TcGlobals) m ty =
+    match tryAppTy g ty with
+    | ValueSome(tcref, [ty]) when isSpanTyconRef g m tcref -> ValueSome(ty)
+    | _ -> ValueNone
+
+let destSpanTy g m ty =
+    match tryDestSpanTy g m ty with
+    | ValueSome(ty) -> ty
+    | _ -> failwith "destSpanTy"
+
+let isReadOnlySpanTyconRef (g: TcGlobals) m (tcref: TyconRef) =
+    if isByrefLikeTyconRef g m tcref then
+        match tcref.ILTyconInfo with
+        | TILObjectReprData(_, _, ilTypeDef) -> ilTypeDef.Name = tname_ReadOnlySpan
```

Comment:
@TIHan.

We usually hold a list of system type refs in TcGlobals which simplifies type comparison. As an example see what we do for ValueTuples:  https://github.com/Microsoft/visualfsharp/blob/a26d32a86c88f4ae61a77bfe5c764cb0ea8cec8b/src/fsharp/TcGlobals.fs#L446

Reply:
We won't be able to use `findSysTyconRef` as that's going to look at the framework tcimports, which Span could exist in System.Memory.dll. I agree we should define it in TcGlobals like this, but will need to find another way.